pkg/api/validation/validation.go

```diff
@@ -4044,7 +4045,7 @@ func validateEndpointPort(port *api.EndpointPort, requireName bool, fldPath *fie
 	} else if len(port.Name) != 0 {
 		allErrs = append(allErrs, ValidateDNS1123Label(port.Name, fldPath.Child("name"))...)
 	}
-	for _, msg := range validation.IsValidPortNum(int(port.Port)) {
+	for _, msg := range validation.IsValidPortNumWithZeroInclusive(int(port.Port)) {
```

Comment:
>For headless services that do not define selectors, the endpoints controller does not create Endpoints records. However, the DNS system looks for and configures either:
CNAME records for ExternalName-type services
A records for any Endpoints that share a name with the service, for all other types

As per https://kubernetes.io/docs/concepts/services-networking/service/#without-selectors, when users configure a headless service without ports, they should create endpoints themselves. If they don't specify ports in endpoint manifest, it's valid (as we remove `if len(ss.Ports) == 0` in L3976). But if they specify ports (with zero value), should we allow this? I might misunderstand here...